### ISSUE TITLE
[stable/jenkins] Missing example for Amazon AWS / S3 backup

### ISSUE THREAD

**Issue by ivandavidov** [helm/charts#15226]
**Describe the bug**
The [backup](https://github.com/helm/charts/tree/master/stable/jenkins#backup) section contains the required properties for Jenkins backup job. However, apart from these properties, there is no example/sample how to apply these properties and what/where to fill in. Since I have no example to use as a starting point, currently I am not able to backup the Jenkins configuration, because I am guessing what to fill in for some of the ``backup`` properties.

**Version of Helm and Kubernetes**:
Not relevant - this is request for better documentation.

**Which chart**:
stable/jenkins

**What happened**:
I was trying to figure out how to properly backup my K8s Jenkins configuration and according to the [Backup](https://github.com/helm/charts/tree/master/stable/jenkins#backup) section I have to provide set of ``backup`` properties. However, there is no example or sample provided and I find it hard to figure out what to fill in.

**What you expected to happen**:
I expected to see clear example (e.g. with dummy AWS credentials) which demonstrates how to use the backup feature. The sample backup configuration be in YAML format.

**How to reproduce it** (as minimally and precisely as possible):
Not relevant - this is request for better documentation.

**Anything else we need to know**:
Guys, you are doing great job! However, please think in advance for the end users who are not familiar with the internal Jenkins chart mechanisms. We really need examples for the different features that are supported by the Jenkins Helm chart.

Keep up the good work!

**Comment by torstenwalter** [comment]
@ivandavidov The backup section you mentioned should describe all necessary values. More details / a yaml example is contained within [values.yaml](https://github.com/helm/charts/blob/master/stable/jenkins/values.yaml#L453-L499)

The [helm documentation](https://helm.sh/docs/helm/#helm-install) explains how to set single values directly and also how to use one/multiple values files.

Which additional information would you like to have?

**Comment by ivandavidov** [comment]
Thank you for the provided feedback. Nevertheless, I still had to reverse-engineer some bits:

* I believe it will be better to explicitly mention that the [existing secret](https://github.com/helm/charts/blob/master/stable/jenkins/values.yaml#L478) is actually K8s secret. Otherwise some people (e.g. myself) could be mistaken that they have to provide the AWS access/secret keys directly.

* You should mention what the default ``kube-tasks`` arguments are. If you have read the backup job definition, then you know the arguments. I read the backup job definition an hour ago. It's not very user friendly to rely on information which is somewhere inside the Helm chart when there is README document in GitHub. I expected to find all relevant information there.

* The backup mechanism doesn't work when the S3 bucket is configured with server-side encryption. The problem is that ``kube-tasks`` rely on ``skbn`` and the latter tool invokes the AWS Go API in very simple way that doesn't provide the argument for automatic server-side encryption. However, this is different issue which I'm going to raise soon.

Long story short - the current backup configuration may work fine for the people who implemented it and/or the people who went through the source configuration of the chart. However, the same backup configuration is quite difficult to handle for people who just want to use it out of the box.

I'll close the issue since I managed to figure out what I need to do, but please, please, for the love of all human beings, when you provide documentation, think about the end users who just want to use the chart out of the box.

Regards,
Ivan Davidov

**Comment by torstenwalter** [comment]
@ivandavidov Sorry if my reaction gave the wrong impression. We'd like to improve the existing documentation wherever it makes sense and simplifies it for users. I really appreciate any contributions and everyone who is spending time to improve the solution. I think they are trying there best to add functionality which might be useful for others as well and document them as good as possible. That's of course sometimes a challenge as the one who creates a PR is often very involved in the solution and it's no always easy to take the view of a consumer.
The same is true for maintainers who review PRs. If you take myself as an example I never worked with the backup functionality myself, but I still feel that it's a useful functionality.

Regarding the concrete improvements you suggest. I really value your feedback! You are probably the best candidate to open a PR and suggest documentation improvements which could be discussed and merged so that others benefit from them.

> I believe it will be better to explicitly mention that the existing secret is actually K8s secret. Otherwise some people (e.g. myself) could be mistaken that they have to provide the AWS access/secret keys directly.

I'd love if you could raise a PR with a documentation update which would have prevented your confusion. 

> You should mention what the default kube-tasks arguments are. If you have read the backup job definition, then you know the arguments. I read the backup job definition an hour ago. It's not very user friendly to rely on information which is somewhere inside the Helm chart when there is README document in GitHub. I expected to find all relevant information there.

I've never been that deep into the backup mechanism. Would be great if you could share your insights as a PR.

> The backup mechanism doesn't work when the S3 bucket is configured with server-side encryption. The problem is that kube-tasks rely on skbn and the latter tool invokes the AWS Go API in very simple way that doesn't provide the argument for automatic server-side encryption. However, this is different issue which I'm going to raise soon.

Same here.

@ivandavidov Please feel encouraged to create PR(s) which improves the documentation.
